logger.js

```diff
@@ -54,17 +61,27 @@ function pinoLogger (opts, stream) {
       return
     }
 
-    log[level]({
-      res: this,
-      responseTime: responseTime
-    }, 'request completed')
+    var shouldLogSuccess = true
+    if (this._currentUrl && autoLoggingOptions && autoLoggingOptions.ignorePaths && autoLoggingOptions.ignorePaths.length) {
+      shouldLogSuccess = autoLoggingOptions.ignorePaths.find(x => this._currentUrl.match(x)) == null
+    }
+
+    if (shouldLogSuccess) {
+      log[level]({
+        res: this,
+        responseTime: responseTime
+      }, 'request completed')
+    }
   }
 
   function loggingMiddleware (req, res, next) {
     req.id = genReqId(req)
     req.log = res.log = logger.child({req: req})
     res[startTime] = res[startTime] || Date.now()
-    if (!req.res) { req.res = res }
+    res._currentUrl = req.url
```

Comment:
I would really _not_ store this here. I would rather run the "should log" check here and change the log level to the child logger to `'error'` or even `'silent'`.

Reply:
I will send the change, for your review. However, I do have one note here. I wanted to keep the current structure where the events (finish, error) get sent anyway because if in the future those event listeners need to execute other logic, then it would not be ran. That is why I wanted to ignore the path at the last moment.

Reply:
But yes, what I can do is to calculate the flag in the middleware, and then just use it in the finish event.

Reply:
On the other hand, how would you go with `silent` level and customLogLevel?

Reply:
about the silent level, I tried this:

```
if (!this.shouldLogSuccess) {
  level = 'silent'
}
log[level](...)
```

and it didn't work. Log object does not have a silent function (and I think in [2018 ](https://github.com/pinojs/pino/issues/434) it was deleted).

Reply:
fixed. waiting for your comments.